tests/integration_tests/functional/test_snapshot_not_losing_dirty_pages.py

```diff
@@ -44,10 +44,12 @@ def test_diff_snapshot_works_after_error(
 
     chroot = Path(uvm.chroot())
 
-    # Create a large file, so we run out of space (ENOSPC) during the snapshot
-    # Assumes a Docker /srv tmpfs of 1G, derived by trial and error
+    # Create a large file dynamically based on available space
     fill = chroot / "fill"
-    subprocess.check_call(f"fallocate -l 330M {fill}", shell=True)
+    free_space = int(subprocess.check_output(f"df {chroot} | tail -1 | awk '{{print $4}}'", shell=True)) # in bytes
```

Comment:
nit: Could use psutil here: 

```
psutil.disk_usage("/tmp")
sdiskusage(
    total=482451128320,
    used=215689019392,
    free=242179715072,
    percent=47.1
)
```

Reply:
This looks much better thanks, I'll update to this